services/dataAccessLayer.js

```diff
@@ -30,7 +30,10 @@ const retrieveUsers = async ({
       result.push(user);
     });
     return result;
-  } else if (userIds.length > 0) {
+  } else if (userIds) {
```

Comment:
why not 
```js 
 else if (userIds.length === 0) {
      return {};
    }

Reply:
userIds can be null in some cases for retrieveUsers functions.

Reply:
Hey @vvaibhavdesai 
I hope the explanation was enough. 
I am resolving this in favor of the previous comment. 
Let me know if you feel otherwise.